src/writer/blocks/writerkeyvaluestorage.py

```
@@ -0,0 +1,84 @@
+                },
+            ),
+        )
+
+    def run(self):
+        from writer.vault import writer_kv_storage
+
+        try:
+            action = self._get_field("action")
+            key = self._get_field("key")
+            if not ALLOWED_CHARS.fullmatch(key):
+                raise ValueError("Key can only contain alphanumeric characters, underscores and hyphens")
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)